client/app/core/rbac.service.js

```diff
@@ -21,4 +30,101 @@ export function RBACFactory() {
   function all() {
     return features;
   }
+
+  function setRBACForNavigation(productFeatures) {
+    var features = {
+      dashboard: {show: entitledForDashboard(productFeatures)},
+      services: {show: entitledForServices(productFeatures)},
+      orders: {show: entitledForServices(productFeatures)},
+      requests: {show: entitledForRequests(productFeatures)},
+      catalogs: {show: entitledForServiceCatalogs(productFeatures)},
+      dialogs: {show: entitledForService(productFeatures)},
+      administration: {show: entitledForService(productFeatures)},
+    };
+
+    return features;
+  }
+  function setNavFeatures(features) {
+    navFeatures = features;
+
+    return navFeatures;
+  }
+  function setActionFeatures(features) {
+    actionFeatures = features;
+
+    return actionFeatures;
+  }
+  function getActionFeatures() {
+    return actionFeatures;
+  }
+  function setRBACForActions(productFeatures) {
+    var features = {
+      serviceView: {show: angular.isDefined(productFeatures.service_view)},
+      serviceEdit: {show: angular.isDefined(productFeatures.service_edit)},
+      serviceTag: {show: angular.isDefined(productFeatures.service_tag)},
+      serviceDelete: {show: angular.isDefined(productFeatures.service_delete)},
+      serviceReconfigure: {show: angular.isDefined(productFeatures.service_reconfigure)},
+      serviceRetireNow: {show: angular.isDefined(productFeatures.service_retire_now)},
+      serviceRetire: {show: angular.isDefined(productFeatures.service_retire)},
+      serviceOwnership: {show: angular.isDefined(productFeatures.service_ownership)},
+    };
+
+    return features;
+  }
+  function getNavFeatures() {
+    return navFeatures;
+  }
+  function entitledForServices(productFeatures) {
+    var serviceFeature = lodash.find(actionFeatures, function(o) {
+      return o.show === true;
+    });
+
+    return angular.isDefined(serviceFeature);
+  }
+
+  function isAnyActionAllowed(actions, productFeatures) {
+    for (var i = 0; i <= actions.length; i++) {
```

Comment:
Doesn't matter all too much since `actions` is a small array but this is a job for `Array#some`.

```javascript
return actions.some(action => angular.isDefined(productFeatures[action]);
```

That way the function always returns a boolean and iteration will stop as soon as possible.